dotcom-rendering/docs/architecture/028-dotenv.md

```diff
@@ -38,3 +38,27 @@ const secrets = [
     // ...
 ];
 ```
+
+## Security
+
+`.env` variables are only available on the server, you should ensure that you **never** pass an environment variable into the clientside code
+e.g
+
+```jsx
+<Island>
+	<!-- Bad :((( -->
+	<MyComponent myProp={process.env.MY_VAR} />
+</Island>
+```
+
+This has considerable security implementations as this variable would then be available to any consumer of the site.
+
+## Limitations
+
+### Server Only
+
+DCR currently only supports `.env` variables being used on the _server_, but not the client.
+
+This limitation exists as the to maintain common & best practice within the organisation, we want to ensure that a re-deploy of the application would involve fetching a fresh environment variable. This means we cannot [inject environment variables through webpack](https://www.npmjs.com/package/dotenv-webpack) (which is done at build time) - and instead we load the dotenv directly when running the server bundle.
```

Comment:
Does the fact we only make this available on the server ensure that we can make changes to env vars on redeploy? Or is it that we run this on start, not build?

Reply:
Both are necessary for this to be possible on redeploy. To make env variables available on the client you need to [inject them through webpack](https://github.com/mrsteele/dotenv-webpack#resulting-bundle), which happens at build time - moving to loading them with the regular `dotenv` package enabled us to use gen-env on start rather than build.

Reply:
Ah, OK I see. I suppose I am getting confused as there are a few points:
* We don't want to expose these values to the client for security reasons (all are considered secrets)
* We want to have these values recyclable on redeploy

I wonder if these aren't limitations but what we expect, and should expect going forward?